src/native/eventpipe/ep-string.c

```diff
@@ -0,0 +1,142 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "ep-rt-config.h"
+
+#ifdef ENABLE_PERFTRACING
+#if !defined(EP_INCLUDE_SOURCE_FILES) || defined(EP_FORCE_INCLUDE_SOURCE_FILES)
+
+#include "ep-rt.h"
+#include <minipal/utf8.h>
+
+static
+ep_char16_t *
+ep_utf8_to_utf16le_string_impl (
+	const ep_char8_t *str,
+	size_t len)
+{
+	if (len == 0) {
+		// Return an empty string if the length is 0
+		ep_char16_t * empty_str = ep_rt_utf16_string_alloc(1);
+		if(empty_str == NULL)
+			return NULL;
+
+		*empty_str = '\0';
+		return empty_str;
+	}
+
+	int32_t flags = MINIPAL_MB_NO_REPLACE_INVALID_CHARS | MINIPAL_TREAT_AS_LITTLE_ENDIAN;
+	size_t ret = minipal_get_length_utf8_to_utf16 (str, len, flags);
+	if (ret <= 0)
+		return NULL;
+
+	ep_char16_t * converted_str = ep_rt_utf16_string_alloc(ret + 1);
+	if (converted_str == NULL)
+		return NULL;
+
+	ret = minipal_convert_utf8_to_utf16 (str, len, (CHAR16_T *)converted_str, ret, flags);
+	converted_str[ret] = '\0';
+	return converted_str;
+}
+
+ep_char16_t *
+ep_rt_utf8_to_utf16le_string (
+	const ep_char8_t *str)
+{
+	if (!str)
+		return NULL;
+
+	return ep_utf8_to_utf16le_string_impl (str, strlen(str));
```

Comment:
On Mono's implementation of ep_utf8_to_utf16le_string_impl had special handling when len was -1 ending up with strlen(str) + 1 as len in g_utf8_to_utf16_impl. Not sure if that was an error introduced in previous changes moving away from Mono's utf8/utf16 conversion, https://github.com/dotnet/runtime/commit/2f2aaae4e4fc1c33d6a7db316fcc0cac4df7af3f, or not or if minipal_get_length_utf8_to_utf16 handle both cases. Actually it looks like that commit changed the meaning of -1 a little since eg_utf8_to_utf16_general  did strlen(str) if len was -1 meaning that this change fixes previous introduced difference in behavior.